In [1]:
import pandas as pd

In [10]:
doc = pd.read_csv("TEDS-A-2014-DS0001-data-excel copy.csv")

In [12]:
doc.head(10)

CASEID  YEAR  AGE  GENDER  RACE  ETHNIC  MARSTAT  EDUC  EMPLOY  \
0  20140000001  2014   12       1     4       5        1     3       4   
1  20140000002  2014    8       2     5       5        4     4       3   
2  20140000003  2014    7       1     5       5        4    -9       4   
3  20140000004  2014    8       1     5       5        1     2       2   
4  20140000005  2014   12       1     5       5        2     4       3   
5  20140000006  2014    7       2     5       5        3     3       4   
6  20140000007  2014    7       1     5       5        1     3       3   
7  20140000008  2014    6       2     5       5        2    -9       3   
8  20140000009  2014    8       1     5       5        2     2       3   
9  20140000010  2014    9       2     5       5        1     3       3   

   DETNLF   ...     BARBFLG  SEDHPFLG  INHFLG  OTCFLG  OTHERFLG  ALCDRUG  \
0       3   ...           0         0       0       0         0        3   
1      -9   ...           0         0       0       0         0        3   
2       3   ...           0         0       0       0         1        3   
3      -9   ...           0         0       0       0         0        3   
4      -9   ...           0         0       0       0         1        3   
5      -9   ...           0         0       0       0         1        2   
6      -9   ...           0         0       0       0         0        2   
7      -9   ...           0         0       0       0         1        2   
8      -9   ...           0         0       0       0         0        3   
9      -9   ...           0         0       0       0         0        2   

   DSMCRIT  PSYPROB  HLTHINS  PRIMPAY  
0       -9        2        2        4  
1        4        2       -9       -9  
2        4        2        4       -9  
3       -9        2        2        4  
4        4        1        3       -9  
5        5        2       -9       -9  
6        5        2       -9        1  
7        5        2        2       -9  
8       -9        2        4        5  
9       -9        2        2        4  

[10 rows x 62 columns]

The file has 10 rows and 62 columns. 

In [14]:
doc = pd.read_csv("TEDS-A-2014-DS0001-data-excel copy.csv", nrows = 10000)

In [16]:
doc.describe

<bound method NDFrame.describe of            CASEID  YEAR  AGE  GENDER  RACE  ETHNIC  MARSTAT  EDUC  EMPLOY  \
0     20140000001  2014   12       1     4       5        1     3       4   
1     20140000002  2014    8       2     5       5        4     4       3   
2     20140000003  2014    7       1     5       5        4    -9       4   
3     20140000004  2014    8       1     5       5        1     2       2   
4     20140000005  2014   12       1     5       5        2     4       3   
5     20140000006  2014    7       2     5       5        3     3       4   
6     20140000007  2014    7       1     5       5        1     3       3   
7     20140000008  2014    6       2     5       5        2    -9       3   
8     20140000009  2014    8       1     5       5        2     2       3   
9     20140000010  2014    9       2     5       5        1     3       3   
10    20140000011  2014   11       1     5       5        4    -9       4   
11    20140000012  2014   10       1     4

File is now limited to 10000 rows

Each row is a case number and the four columns are the Case Id, the age of person receiving treatment, their gender, and their race. 

In [27]:
doc.RACE.describe(), doc.GENDER.describe(), doc.CASEID.describe(), doc.AGE.describe()

(count    10000.000000
 mean         5.546200
 std          3.577953
 min         -9.000000
 25%          5.000000
 50%          5.000000
 75%          5.000000
 max         23.000000
 Name: RACE, dtype: float64, count    10000.000000
 mean         1.384300
 std          0.486454
 min          1.000000
 25%          1.000000
 50%          1.000000
 75%          2.000000
 max          2.000000
 Name: GENDER, dtype: float64, count    1.000000e+04
 mean     2.014001e+10
 std      2.886896e+03
 min      2.014000e+10
 25%      2.014000e+10
 50%      2.014001e+10
 75%      2.014001e+10
 max      2.014001e+10
 Name: CASEID, dtype: float64, count    10000.000000
 mean         7.232400
 std          2.493113
 min          2.000000
 25%          5.000000
 50%          7.000000
 75%          9.000000
 max         12.000000
 Name: AGE, dtype: float64)

For Race, the max is 23 and the min is -9, which indicates Hawaians and Unknown, respecitively. For Gender, the max was 2 and min was 1, indicating female and male, respecitively. For CASE ID the max is 12 and min is 2. 

In [42]:
doc.SUB1.groupby(doc.AGE).value_counts().head(130)

AGE  SUB1
2     4       72
      2       13
      7        2
      13       2
     -9        1
      1        1
      17       1
      20       1
3     4      422
      2       66
      10      18
      7       14
      20      10
      5        8
      11       5
      1        4
      3        4
      9        4
      13       4
      18       4
     -9        2
      6        2
      16       2
4     4      278
      2      142
      5       71
      10      42
      7       35
      13      16
      20      16
            ... 
8     18       1
9     2      424
      5       97
      10      91
      4       69
      7       68
      3       66
      20      11
      13       7
     -9        5
      1        4
      8        4
      11       2
      9        1
      16       1
10    2      385
      3      100
      5       70
      4       63
      10      47
      7       46
      13       5
      20       4
     -9        3
      6        3
      11       2
      18       1
11  

Top three substances abused by those younger than 18 are marijuana, alcohol, and methaphetamine

The top three substances abused by those older than 50 are alcohol, heroin, and cocain/crack.

The top three substances abused by those between 18 and 50 are alcohol, heroin, and meth.
4:(18-20) [4,2,5]; 5:(21-24) [6,11,1]; 6:(25-29) [2,5,4]; 7:(30-34) [2,5,10]; 8:(35-39) [2,5,10]; 9:(40-44) [2,5,10]; 10:(45-49) [7,13,20]

Question 8, don't understand what it is asking for... 

In [50]:
doc.STFIPS.value_counts()

18    3546
26    1793
29    1165
55    1106
20     877
17     587
39     472
27     302
38      64
19      61
46       6
36       5
53       2
31       2
41       2
40       1
2        1
10       1
12       1
42       1
11       1
44       1
6        1
22       1
24       1
Name: STFIPS, dtype: int64

The state with the largest admittances was Indiana with 3546 

In [58]:
doc.CBSA.groupby(doc.PSOURCE==7).value_counts().head(500)

PSOURCE  CBSA  
False    -9        3165
          19820     608
          28140     248
          33460     187
          26900     174
          48620     169
          16980     161
          41180     135
          24340     124
          22420     110
          19380      84
          34740      72
          40980      60
          27900      58
          44180      55
          45820      48
          28020      46
          17140      36
          17860      32
          29620      32
          12980      29
          45900      28
          18140      26
          27620      24
          17460      22
          37900      14
          41140      13
          10420      12
          41060      12
          43780      12
                   ... 
True      44100       8
          19340       7
          20260       7
          26980       7
          36860       7
          37900       6
          17860       5
          27620       5
          49660       5
          13020       4


The Denver area is more likely to be referred through a court order. 

In [66]:
doc.CBSA.groupby(doc.SUB1==5).value_counts()

SUB1   CBSA  
False  -9        5496
        19820     403
        28140     375
        48620     275
        16980     205
        33460     184
        24340     143
        41180     132
        22420     113
        26900     106
        27900      85
        19380      82
        45820      82
        34740      77
        44180      72
        45900      72
        40980      53
        12980      41
        28020      41
        29620      38
        18140      34
        17860      32
        17460      31
        17140      29
        27620      27
        43780      19
        14010      18
        41140      17
        20260      16
        37900      16
                 ... 
True    48620       6
        17860       5
        45900       5
        19500       4
        37900       4
        10420       3
        33780       3
        43780       3
        44100       3
        49660       3
        14010       2
        19340       2
        20260       2
        27620     

It looks like New York will be more likely to use heroin.

In [69]:
doc.SUB1.groupby(doc.DSMCRIT==15).value_counts()

DSMCRIT  SUB1
False     2      3660
          4      2020
          5      1457
          7       964
          10      929
          3       471
          13      131
          20      114
         -9        45
          11       44
          6        41
          1        26
          16       17
          8        15
          9        15
          18       12
          17        7
          15        3
          12        2
          14        1
True      2        12
          1         9
          4         4
          5         1
Name: SUB1, dtype: int64

Depressives usually go for treatment of alcohol, nothing, and marijuana. 

In [72]:
doc.SUB1.groupby(doc.DAYWAIT).value_counts()

DAYWAIT  SUB1
-9        2      2039
          4       981
          5       637
          7       513
          10      400
          3       216
          20       93
          13       83
         -9        45
          11       23
          6        11
          16        9
          9         7
          1         6
          18        6
          8         2
          12        1
          14        1
          15        1
 0        2       911
          4       617
          10      385
          5       384
          7       209
          3       149
          13       20
          1        13
          11       12
          20        9
          6         8
                 ... 
 604      3         1
 630      2         1
 640      4         1
 655      3         1
 672      5         1
 675      7         1
 700      2         5
          4         3
          7         3
          5         2
          3         1
 704      4         1
 710      2         1
 721      3       

Heroin treatment has the longest waiting time.

In [78]:
doc.SUB1.groupby(doc.REGION).value_counts()

REGION  SUB1
1        5        15
         7         9
         2         8
         4         8
         3         3
         10        2
         6         1
         8         1
         13        1
         16        1
         20        1
2        2      3662
         4      2016
         5      1442
         7       951
         10      926
         3       468
         13      130
         20      112
        -9        45
         11       44
         6        39
         1        34
         16       16
         9        15
         8        13
         18       12
         17        7
         15        3
         12        2
         14        1
3        2         2
         1         1
         6         1
         7         1
         8         1
4        7         3
         5         1
         10        1
         20        1
Name: SUB1, dtype: int64

The Midwest and South suffer from alcohol substance abuse the most, while those in the west suffer from other opiates, and those in the NorthEast suffer from heroin. Also, the NorthEast has far more cases than the other census regions, while the MidWest has the least.  

In [85]:
doc.SUB1.groupby(doc.ARRESTS).value_counts()
doc.REGION.groupby(doc.ARRESTS).value_counts()
doc.PSOURCE.groupby(doc.ARRESTS).value_counts()

ARRESTS  PSOURCE
-9        7          151
          1           59
          3           14
          6            8
         -9            5
          2            5
 0        7         3450
          1         2819
          6         1036
          3          631
          2          606
         -9           85
          4           58
          5           32
 1        7          383
          1          238
          6           94
          2           38
          3           32
         -9           20
          4            4
 2        1          116
          7           75
          6           26
          3            7
          2            4
         -9            2
          4            2
Name: PSOURCE, dtype: int64

Most arrests occured more than once due to alcoholism or heroin addiction. Arrests usually occured because of alcoholism. 

Most arrests took place in the Midwest. 

In cases of rearrest or intial arrest of an individual the referral to a treatment program was made by the individual or a court. 

1. What substance led to the most arrests?
2. What region had the longest wait time?
3. What state had the most criminal justice referrals? 
4. What state had the longest wait times?
5. What state had the most arrests?